### PR TITLE
bug fixes in publish cc

### PR DESCRIPTION
**Pull request by AshwiniChalla**

### CODE REVIEW

**Comment by nigurr** [comment]
Issue Details and fix

**Comment by AshwiniChalla** [comment]
2 bugs: 
1. code coverage fields should show like title case. Currently it is shown like whatever given in summary xml.
2. If coverage numbers are not integers, those numbers are not being published.
Fixed these 2 issue and added unit tests. Done manual testing.

File: src/agent/codecoveragesummaryreader.ts [hunk 1]
```diff
@@ -144,15 +152,31 @@ export class CoberturaSummaryReader implements ccp.ICodeCoverageReader {
...
+                        return defer.promise;
+                    }
                 }
 
                 if (branchesCovered && branchesTotal) {
-                    this.command.info("Branches : " + branchesCovered + "/" + branchesTotal + " covered.");
-                    var coverageStat = SummaryReaderUtilities.getCodeCoverageStatistics("Branches", branchesCovered, branchesTotal, "branch");
-                    coverageStats.push(coverageStat);
+                    branchesCovered = parseInt(branchesCovered);
+                    branchesTotal = parseInt(branchesTotal);
```
Comment (nigurr) [inline src/agent/codecoveragesummaryreader.ts]: format code and remove extra whitespaces

File: src/agent/utilities.ts [hunk 2]
```diff
@@ -15,6 +15,14 @@ export interface GetOrCreateResult<T> {
     created: boolean;
     result: T;
 }
+
+//converts inputString to titleCase string. For example, "abc def" is converted to "Abc Def"
+export function toTitleCase(inputString: string): string {
+    if (inputString) {
+        return inputString.replace(/\w\S*/g, function(str) { return str.charAt(0).toUpperCase() + str.substr(1).toLowerCase(); });
```
Comment (nigurr) [inline src/agent/utilities.ts]: Test cases?

**Comment by nigurr** [comment]
Approved with suggestions.

File: src/agent/codecoveragesummaryreader.ts [hunk 3]
```diff
@@ -70,10 +70,18 @@ export class JacocoSummaryReader implements ccp.ICodeCoverageReader {
                 var counterNode = reportNode.counter.at(i);
                 var attributes = counterNode.attributes();
                 if (attributes && attributes.type && attributes.covered && attributes.missed) {
-                    var total = Number(attributes.covered) + Number(attributes.missed);
-                    this.command.info(attributes.type + " : " + attributes.covered + "/" + total + " covered.");
-                    var coverageStat = SummaryReaderUtilities.getCodeCoverageStatistics(attributes.type, attributes.covered, total, attributes.type);
-                    coverageStats.push(coverageStat);
+                    var covered = parseInt(attributes.covered);
+                    var missed = parseInt(attributes.missed);
+                    if (!isNaN(covered) && !isNaN(missed)) {
```
Comment (nigurr) [inline src/agent/codecoveragesummaryreader.ts]: code format everywhere and remove whitespaces

File: src/agent/utilities.ts (hunk 2 above)
Comment (AshwiniChalla) [inline src/agent/utilities.ts]: This test is covered in codeCoveredPublisher tests.

Comment (nigurr) [inline src/agent/utilities.ts]: If you are writing utility method, it should be independent. It needs it's own UT

Comment (AshwiniChalla) [inline src/agent/utilities.ts]: Added

**Pull request merged**